### PR TITLE
 Modernize Python 2 code to get ready for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
* __xrange()__ was removed in Python 3 in favor of __range()__.
* Python 3 does not allow explicit tuples as function parameters.

### CODE REVIEW

File: dynamic_batching_test.py [hunk 1]
```diff
@@ -26,6 +26,11 @@
 
 import tensorflow as tf
 
+try:
```
Comment (lespeholt) [inline dynamic_batching_test.py]: The default way in TensorFlow is:
from six.moves import range
We should do the same.

Comment (lespeholt) [inline dynamic_batching_test.py]: The default way in TensorFlow is:
from six.moves import range
We should do the same.

File: experiment.py [hunk 2]
```diff
@@ -246,8 +253,10 @@ def create_state(t):
       create_state, (initial_env_state, initial_env_output, initial_agent_state,
                      initial_agent_output))
 
-  def step((env_state, env_output, agent_state, agent_output), unused_i):
+  def step(env_state__env_output__agent_state__agent_output, unused_i):
```
Comment (lespeholt) [inline experiment.py]: I think it's cleaner just to call it input_.

**Comment by cclauss** [comment]
I originally avoided __six__ because it is nowhere else in the codebase.  It is in the mods.

File: experiment.py [hunk 3]
```diff
@@ -194,7 +197,8 @@ def _head(self, core_output):
 
     return AgentOutput(new_action, policy_logits, baseline)
 
-  def _build(self, (action, env_output), core_state):
+  def _build(self, action__env_output, core_state):
```
Comment (lespeholt) [inline experiment.py]: input_

File: experiment.py [hunk 4]
```diff
@@ -138,7 +140,8 @@ def _instruction(self, instruction):
     # Return last output.
     return tf.reverse_sequence(output, length, seq_axis=1)[:, 0]
 
-  def _torso(self, (last_action, env_output)):
+  def _torso(self, last_action__env_output):
```
Comment (lespeholt) [inline experiment.py]: input_

File: experiment.py [hunk 5]
```diff
@@ -37,6 +37,8 @@
 except tf.errors.NotFoundError:
   tf.logging.warning('Running without dynamic batching.')
 
+from six.moves import range
```
Comment (lespeholt) [inline experiment.py]: lambda t: tf.transpose(t, [1, 0] + range(t.shape.ndims)[2:]), s)

now fail with:

TypeError: sequence index must be integer, not 'slice'

File: experiment.py [hunk 6]
```diff
@@ -246,8 +250,9 @@ def create_state(t):
       create_state, (initial_env_state, initial_env_output, initial_agent_state,
                      initial_agent_output))
 
-  def step((env_state, env_output, agent_state, agent_output), unused_i):
+  def step(input_, unused_i):
     """Steps through the agent and the environment."""
+    (env_state, env_output, agent_state, agent_output) = input_
```
Comment (lespeholt) [inline experiment.py]: Remove ( and )

File: experiment.py (hunk 5 above)
Comment (cclauss) [reply experiment.py]: Fixed with __list(range())__.

**Pull request merged**